### PR TITLE
Fix #2714: Allow setting security context of pods using persistent volumes

### LINKED ISSUE #2714: Allow setting security context of pods using persistent volumes
**Overview**

In order to support Ceph persistent volume claims, pods may need to be configured with a security context. This affects components such as:

* standard authservice
* brokered address space broker
* standard address space broker

For the brokers, the podTemplate.spec in the standardinfraconfig/brokeredinfraconfig can be extended to specify the security context.

For standard-authservice, the syntax of AuthenticationService resource needs to be extended.

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

- Enhancement / new feature

### Description

Allows the security context of pods that use a persistent volume to be specified by
way of the CR that controls the resource.

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [x] Update CHANGELOG.md

### CODE REVIEW

File: pkg/apis/iot/v1alpha1/types_config.go [hunk 1]
```diff
@@ -116,6 +116,7 @@ type OperatorConfig struct {
 }
 
 type DeviceRegistryServiceConfig struct {
+	// Add to here?
```
Comment (lulf) [inline pkg/apis/iot/v1alpha1/types_config.go]: Remove comment?

File: pkg/apis/iot/v1alpha1/types_config.go [hunk 2]
```diff
@@ -148,6 +149,7 @@ type ExternalCacheNames struct {
 type CommonServiceConfig struct {
 	Container *ContainerConfig      `json:"container,omitempty"`
 	Java      *JavaContainerOptions `json:"java,omitempty"`
+	corev1.SecurityContext
```
Comment (lulf) [inline pkg/apis/iot/v1alpha1/types_config.go]: Is this needed when it is set on the device registry config itself?

File: pkg/controller/authenticationservice/standard.go [hunk 3]
```diff
@@ -101,14 +101,12 @@ func applyStandardAuthServiceDeployment(authservice *adminv1beta1.Authentication
...
-	}
-
-	err = install.ApplyFsGroupOverride("FS_GROUP_STANDARD_AUTHSERVICE_OVERRIDE", deployment)
-	if err != nil {
-		return err
+	if authservice.Spec.Standard.SecurityContext == nil {
+		log.Info("KWDEBUG hello nil")
+		deployment.Spec.Template.Spec.SecurityContext = nil
+	} else {
+		log.Info("KWDEBUG hello non-nil", "sc", authservice.Spec.Standard.SecurityContext)
```
Comment (lulf) [inline pkg/controller/authenticationservice/standard.go]: debug logging.

File: pkg/apis/iot/v1alpha1/types_config.go (hunk 2 above)
Comment (k-wall) [reply pkg/apis/iot/v1alpha1/types_config.go]: It was the remains of an earlier idea. Removed.

File: pkg/controller/authenticationservice/standard.go (hunk 3 above)
Comment (k-wall) [reply pkg/controller/authenticationservice/standard.go]: Gone

**Comment by k-wall** [comment]
@lulf can re-review.  The code should now work out of the box when installed in openshift-operators (well, except CRC - https://github.com/code-ready/crc/issues/814)

**Comment by famartinrh** [comment]
@enmasse-ci run tests testcase=OLMSpecificTest ocp4 OLM

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=smoke

**Comment by k-wall** [comment]
@enmasse-ci run tests profile=smoke

**Pull request merged**